File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/tasks/TasksApi.java
package co.uk.swarmbit.docker.api.tasks;

import co.uk.swarmbit.docker.api.services.parameters.ServiceLogsParameters;
import co.uk.swarmbit.docker.api.tasks.parameters.TasksListParameters;
import co.uk.swarmbit.docker.api.common.json.TaskJson;

import java.util.List;

public interface TasksApi {

    List<TaskJson> listTasks(String swarmId, TasksListParameters parameters);

    TaskJson inspectTask(String swarmId, String id);

    byte[] getTaskLogs(String swarmId, String id, ServiceLogsParameters parameters);

}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/tasks/TasksApi.kt
package com.swarmbit.docker.api.tasks

import com.swarmbit.docker.api.services.parameters.ServicesLogsParameters
import com.swarmbit.docker.api.tasks.model.Task
import com.swarmbit.docker.api.tasks.parameters.TasksListParameters

interface TasksApi {
    fun listTasks(swarmId: String, parameters: TasksListParameters): List<Task>
    fun inspectTask(swarmId: String, id: String): Task
    fun getTaskLogs(swarmId: String, id: String, parameters: ServicesLogsParameters): ByteArray
}
